upgrade/sql/8.1.8-catchup.sql

```diff
@@ -0,0 +1,3 @@
+/* script intended for catching invalid requests in previous scripts */
+
+ALTER TABLE `PREFIX_authorized_application` CHANGE `name` `name` VARCHAR(255) NOT NULL;
```

Comment:
Is this PR not doing what we said we would not do in https://github.com/PrestaShop/autoupgrade/pull/650#issuecomment-1914307091 ? 😄

Reply:
This fixes an error from a previous script, to be on par with a new installation. It's catching up

Reply:
the same way as was done on the 8.1.7-catchup.php script

Reply:
I am removing it because this table will be deleted in v9

Reply:
If there was an issue in migration script from 8.0.0 to 8.1.0 we should fix the 8.1.0.sql migration script, not the 8.1.7 migration script.

Similarly if there was an issue in migration script from 1.7.8.11 to 1.7.8.12 we should fix the 1.7.8.12.sql migration script, not the 8.1.8 migration script, no? 😉 

I think it's similar to https://github.com/PrestaShop/autoupgrade/pull/650#issuecomment-1914307091 because you have the same thinking "well, people already did the 8.1 upgrade so if I want them to apply these changes we forgot, I'll slip them into next upgrade"

I think this is a right problem but a wrong solution 😉 . What about a different system?

An idea, to be explored: everytime we run a SQL migration script, let's store the info in MySQL. User John Doe runs 8.0.0.sql script -> we mark in a SQL table "he's run the 8.0.0.sql script". User John Doe runs 8.1.0.sql script -> we mark in a SQL table "he's run the 8.1.0.sql script". Something that Doctrine Migration does (marking which script was performed, and which was not).

Then we enable the possibility to create catch-up files 😉 and allow user to run them.

In that situation, if we find out we forgot some changes in 8.1.0 upgrade process, we put the forgotten changes into a 8.1.0-catchup1.sql file. If we find out we forgot some changes in 8.1.1 upgrade process, we put the forgotten changes into a 8.1.1-catchup1.sql file. And so on.

This way 😉 we keep the 8.1.7.sql script "clean" as it contains the changes to alter the database and data between 8.1.6 and 8.1.7 . And only that. No extra 😄 .

The catch-ups system I just suggested could also be modified to include scripts suggested by @ShaiMagal in https://github.com/PrestaShop/autoupgrade/pull/650

Reply:
@matks 

We have already talked about migrations in the same spirit as that of doctrine, and this is a much better approach. However, this would require an overhaul of the current system, which is not currently planned in the current module. We therefore proposed a simple and quick solution to "catch up" on errors from previous scripts.